src/Nethermind/Nethermind.JsonRpc/JsonRpcProcessor.cs

```diff
@@ -91,46 +91,67 @@ private IEnumerable<(JsonRpcRequest Model, List<JsonRpcRequest> Collection)> Des
             {
                 if (token is JArray array)
                 {
-                    foreach (JToken tokenElement in array)
-                    {
-                        UpdateParams(tokenElement);
-                    }
-
-                    yield return (null, array.ToObject<List<JsonRpcRequest>>(_obsoleteBasicJsonSerializer));
+                    yield return (null, array.Select(JTokenToJsonRpcRequest).ToList());
                 }
                 else
                 {
-                    UpdateParams(token);
-                    yield return (token.ToObject<JsonRpcRequest>(_obsoleteBasicJsonSerializer), null);
+                    yield return (JTokenToJsonRpcRequest(token), null);
                 }
             }
         }
 
-        private void UpdateParams(JToken token)
+        /// <summary>
+        /// Manually convert JToken to JsonRpcRequests as the param inside JsonRpcRequests is a JToken. If we use
+        /// a JTokenReader, it would iterate through the JSON which is a waste of time, as we already have a JToken.
+        /// </summary>
+        /// <param name="token"></param>
+        /// <returns></returns>
+        private JsonRpcRequest JTokenToJsonRpcRequest(JToken token)
         {
-            var paramsToken = token.SelectToken("params");
-            if (paramsToken is null)
+            JsonRpcRequest jsonRpcRequest = new JsonRpcRequest();
+
+            JToken? jsonRpcField = token["jsonrpc"];
+            if (jsonRpcField != null)
+            {
+                jsonRpcRequest.JsonRpc = jsonRpcField.Value<string>();
+            }
+
+            JToken? methodField = token["method"];
+            if (methodField != null)
+            {
+                jsonRpcRequest.Method = methodField.Value<string>();
+            }
+
+            JToken? idField = token["id"];
+            if (idField != null)
             {
-                paramsToken = token.SelectToken("Params");
-                if (paramsToken is null)
+                // Not sure what is the logic here.. not gonna unuse the IdConverter
+                JTokenReader reader = new JTokenReader(idField);
+                if (reader.Read())
                 {
-                    return;
+                    jsonRpcRequest.Id = new IdConverter().ReadJson(reader, null, null, null);
                 }
             }
 
-            if (paramsToken is JValue)
+            JToken? paramsField = token["params"];
```

Comment:
Should we check for "Jsonrpc"/"JsonRpc"/"Id"/etc too?
Btw, can be `token["params"] ?? token["params"]`

Reply:
I'm confused as to why we check the capitalized case in the first place. The spec mentioned them all as small case. https://www.jsonrpc.org/specification. Should we check regardless?

Reply:
I'll just check it regardless.